packages/insomnia/src/ui/components/viewers/response-multipart-viewer.tsx

```diff
@@ -158,157 +125,148 @@ export class ResponseMultipartViewer extends PureComponent<Props, State> {
-      disableHtmlPreviewJs,
-      disablePreviewLinks,
-      editorFontSize,
-      filter,
-      filterHistory,
-      responseId,
-      url,
-    } = this.props;
-    const { activePart, parts, error } = this.state;
-
-    if (error) {
-      return (
-        <div
-          className="pad monospace"
-          style={{
-            fontSize: editorFontSize,
-          }}
-        >
-          Failed to parse multipart response: {error}
-        </div>
-      );
-    }
-
-    const selectedPart = parts[activePart];
-    return (
+  if (parts.length === 0 || !selectedPart) {
+    return null;
+  }
+  return (
+    <div
+      className="pad-sm tall wide"
+      style={{
+        display: 'grid',
+        gridTemplateRows: 'auto minmax(0, 1fr)',
+      }}
+    >
       <div
-        className="pad-sm tall wide"
+        className="pad-bottom-sm"
         style={{
           display: 'grid',
-          gridTemplateRows: 'auto minmax(0, 1fr)',
+          gridTemplateColumns: 'minmax(0, 1fr) auto',
         }}
       >
-        <div
-          className="pad-bottom-sm"
-          style={{
-            display: 'grid',
-            gridTemplateColumns: 'minmax(0, 1fr) auto',
-          }}
-        >
-          <div>
-            <Dropdown wide>
-              <DropdownButton className="btn btn--clicky">
-                <div
-                  style={{
-                    minWidth: '200px',
-                    display: 'inline-block',
-                  }}
-                >
-                  {selectedPart ? this._describePart(selectedPart) : 'Unknown'}
-                </div>
-                <i className="fa fa-caret-down fa--skinny space-left" />
-              </DropdownButton>
-              {parts.map((part, i) => (
-                <DropdownItem key={i} value={i} onClick={this._handleSelectPart}>
-                  {i === activePart ? <i className="fa fa-check" /> : <i className="fa fa-empty" />}
-                  {this._describePart(part)}
-                </DropdownItem>
-              ))}
-            </Dropdown>
-          </div>
-          <Dropdown right>
+        <div>
+          <Dropdown wide>
             <DropdownButton className="btn btn--clicky">
-              <i className="fa fa-bars" />
+              <div
+                style={{
+                  minWidth: '200px',
+                  display: 'inline-block',
+                }}
+              >
+                {selectedPart.title}
+              </div>
+              <i className="fa fa-caret-down fa--skinny space-left" />
             </DropdownButton>
-            <DropdownItem onClick={this._handleViewHeaders}>
-              <i className="fa fa-list" /> View Headers
-            </DropdownItem>
-            <DropdownItem onClick={this._handleSaveAsFile}>
-              <i className="fa fa-save" /> Save as File
-            </DropdownItem>
+            {parts.map(part => (
+              <DropdownItem key={part.title} value={part} onClick={selectPart}>
```

Comment:
We could use part.id instead
```suggestion
              <DropdownItem key={part.id} value={part} onClick={selectPart}>
```